## Importação de Bibliotecas e Carregamento dos Dados

In [17]:
import pandas as pd

df = pd.read_csv("../data/customer_support_tickets.csv")

def print_formatado(*args, prefixo=5, largura_total=100):
    titulo = " ".join(str(a) for a in args)
    print("\n")
    print("=" * 100)
    print(titulo)
    print("=" * 100)

## Inspeção Inicial

In [18]:
print_formatado("1. Tamanho do dataset:")
print(df.shape)

print_formatado("2. Variáveis do dataset:")
print(df.columns)

print_formatado("3. Significado das colunas:")
print("Ver Markdown")

print_formatado("4. Tipos dos dados:")
print(df.dtypes)

print_formatado("5. Organização Registros:")
df.info()
print("Ver Markdown")

print_formatado("6. Exemplo de valores:")
print("\nPrimeiros registros:")
print(df.head())

print("\nÚltimos registros:")
print(df.tail())

print("\nRegistros aleatórios:")
print(df.sample(5, random_state=1))

print("\nExemplo dataset:")
display(df)


print_formatado("7. Identificação das variáveis")
print("\nVariáveis numéricas:")
print(f"Ticket ID: min={df['Ticket ID'].min()}, max={df['Ticket ID'].max()}, média={df['Ticket ID'].mean():.2f}")
print(f"Customer Age: min={df['Customer Age'].min()}, max={df['Customer Age'].max()}, média={df['Customer Age'].mean():.2f}")
print(f"Customer Satisfaction Rating: min={df['Customer Satisfaction Rating'].min()}, max={df['Customer Satisfaction Rating'].max()}, média={df['Customer Satisfaction Rating'].mean():.2f}")

print("\nVariáveis categóricas:")
print(f"Customer Gender: {df['Customer Gender'].unique().tolist()}")
print(f"Product Purchased: {df['Product Purchased'].unique().tolist()}")
print(f"Ticket Type: {df['Ticket Type'].unique().tolist()}")
print(f"Ticket Subject: {df['Ticket Subject'].unique().tolist()}")
print(f"Ticket Status: {df['Ticket Status'].unique().tolist()}")
print(f"Ticket Priority: {df['Ticket Priority'].unique().tolist()}")
print(f"Ticket Channel: {df['Ticket Channel'].unique().tolist()}")

print("\nVariáveis textuais/identificadoras:")
print("Ver Markdown")

print("\nVariáveis de data/hora:")
print("Ver Markdown")



1. Tamanho do dataset:
(8469, 17)


2. Variáveis do dataset:
Index(['Ticket ID', 'Customer Name', 'Customer Email', 'Customer Age',
       'Customer Gender', 'Product Purchased', 'Date of Purchase',
       'Ticket Type', 'Ticket Subject', 'Ticket Description', 'Ticket Status',
       'Resolution', 'Ticket Priority', 'Ticket Channel',
       'First Response Time', 'Time to Resolution',
       'Customer Satisfaction Rating'],
      dtype='str')


3. Significado das colunas:
Ver Markdown


4. Tipos dos dados:
Ticket ID                         int64
Customer Name                       str
Customer Email                      str
Customer Age                      int64
Customer Gender                     str
Product Purchased                   str
Date of Purchase                    str
Ticket Type                         str
Ticket Subject                      str
Ticket Description                  str
Ticket Status                       str
Resolution                          str
Ticket

,Ticket ID,Customer Name,Customer Email,Customer Age,Customer Gender,Product Purchased,Date of Purchase,Ticket Type,Ticket Subject,Ticket Description,Ticket Status,Resolution,Ticket Priority,Ticket Channel,First Response Time,Time to Resolution,Customer Satisfaction Rating
0,1,Marisa Obrien,carrollallison@example.com,32,Other,GoPro Hero,2021-03-22,Technical issue,Product setup,I'm having an issue with the {product_purchase...,Pending Customer Response,NaN,Critical,Social media,2023-06-01 12:15:36,NaN,NaN
1,2,Jessica Rios,clarkeashley@example.com,42,Female,LG Smart TV,2021-05-22,Technical issue,Peripheral compatibility,I'm having an issue with the {product_purchase...,Pending Customer Response,NaN,Critical,Chat,2023-06-01 16:45:38,NaN,NaN
2,3,Christopher Robbins,gonzalestracy@example.com,48,Other,Dell XPS,2020-07-14,Technical issue,Network problem,I'm facing a problem with my {product_purchase...,Closed,Case maybe show recently my computer follow.,Low,Social media,2023-06-01 11:14:38,2023-06-01 18:05:38,3.0
3,4,Christina Dillon,bradleyolson@example.org,27,Female,Microsoft Office,2020-11-13,Billing inquiry,Account access,I'm having an issue with the {product_purchase...,Closed,Try capital clearly never color toward story.,Low,Social media,2023-06-01 07:29:40,2023-06-01 01:57:40,3.0
4,5,Alexander Carroll,bradleymark@example.com,67,Female,Autodesk AutoCAD,2020-02-04,Billing inquiry,Data loss,I'm having an issue with the {product_purchase...,Closed,West decision evidence bit.,Low,Email,2023-06-01 00:12:42,2023-06-01 19:53:42,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
8464,8465,David Todd,adam28@example.net,22,Female,LG OLED,2021-12-08,Product inquiry,Installation support,My {product_purchased} is making strange noise...,Open,NaN,Low,Phone,NaN,NaN,NaN
8465,8466,Lori Davis,russell68@example.com,27,Female,Bose SoundLink Speaker,2020-02-22,Technical issue,Refund request,I'm having an issue with the {product_purchase...,Open,NaN,Critical,Email,NaN,NaN,NaN
8466,8467,Michelle Kelley,ashley83@example.org,57,Female,GoPro Action Camera,2021-08-17,Technical issue,Account access,I'm having an issue with the {product_purchase...,Closed,Eight account century nature kitchen.,High,Social media,2023-06-01 09:44:22,2023-06-01 04:31:22,3.0
8467,8468,Steven Rodriguez,fpowell@example.org,54,Male,PlayStation,2021-10-16,Product inquiry,Payment issue,I'm having an issue with the {product_purchase...,Closed,We seat culture plan.,Medium,Email,2023-06-01 18:28:24,2023-06-01 05:32:24,3.0




7. Identificação das variáveis

Variáveis numéricas:
Ticket ID: min=1, max=8469, média=4235.00
Customer Age: min=18, max=70, média=44.03
Customer Satisfaction Rating: min=1.0, max=5.0, média=2.99

Variáveis categóricas:
Customer Gender: ['Other', 'Female', 'Male']
Product Purchased: ['GoPro Hero', 'LG Smart TV', 'Dell XPS', 'Microsoft Office', 'Autodesk AutoCAD', 'Microsoft Surface', 'Philips Hue Lights', 'Fitbit Versa Smartwatch', 'Dyson Vacuum Cleaner', 'Nintendo Switch', 'Microsoft Xbox Controller', 'Nintendo Switch Pro Controller', 'Nest Thermostat', 'Sony PlayStation', 'GoPro Action Camera', 'Xbox', 'LG Washing Machine', 'Canon EOS', 'HP Pavilion', 'Amazon Kindle', 'Lenovo ThinkPad', 'Fitbit Charge', 'Adobe Photoshop', 'Google Pixel', 'Amazon Echo', 'PlayStation', 'Samsung Galaxy', 'iPhone', 'LG OLED', 'Sony Xperia', 'Apple AirPods', 'Sony 4K HDR TV', 'Canon DSLR Camera', 'Roomba Robot Vacuum', 'Nikon D', 'Bose QuietComfort', 'Samsung Soundbar', 'Asus ROG', 'Bose SoundLink Speak

### Análise da Inspeção Inicial

**3. Significado das colunas**

| Coluna | Descrição |
|---|---|
| `Ticket ID` | Identificador único e sequencial do chamado. |
| `Customer Name` | Nome do cliente que abriu o chamado. |
| `Customer Email` | E-mail do cliente (dado de contato/PII). |
| `Customer Age` | Idade do cliente em anos. |
| `Customer Gender` | Gênero informado pelo cliente (`Male`, `Female`, `Other`). |
| `Product Purchased` | Produto ao qual o chamado se refere (42 produtos distintos). |
| `Date of Purchase` | Data em que o produto foi comprado. |
| `Ticket Type` | Categoria do chamado (ex.: `Technical issue`, `Billing inquiry`). |
| `Ticket Subject` | Assunto específico dentro do tipo de chamado (16 assuntos). |
| `Ticket Description` | Texto livre descrevendo o problema relatado pelo cliente. |
| `Ticket Status` | Situação atual do chamado (`Open`, `Pending Customer Response`, `Closed`). |
| `Resolution` | Texto livre com a solução aplicada — só existe para chamados `Closed`. |
| `Ticket Priority` | Prioridade atribuída ao chamado (`Low`, `Medium`, `High`, `Critical`). |
| `Ticket Channel` | Canal de abertura do chamado (`Email`, `Phone`, `Chat`, `Social media`). |
| `First Response Time` | Data/hora da primeira resposta da equipe de suporte. |
| `Time to Resolution` | Data/hora em que o chamado foi resolvido — só existe para `Closed`. |
| `Customer Satisfaction Rating` | Nota de satisfação (1 a 5) — só existe para chamados `Closed`. |

**5. Organização dos registros**

Cada linha representa **um chamado de suporte** (grão = ticket), sem hierarquia entre linhas. O `df.info()` mostra 8.469 registros e 17 colunas, majoritariamente do tipo `object` (texto/categoria), com `Ticket ID` e `Customer Age` como `int64` e `Customer Satisfaction Rating` como `float64` (por conter nulos). Os tipos batem com a natureza de cada coluna — nenhuma conversão incoerente aparente nesta etapa.

**Variáveis textuais/identificadoras**

- `Ticket ID`, `Customer Name`, `Customer Email`: identificam o registro/cliente, não devem entrar em estatísticas ou modelos como feature.
- `Ticket Description`, `Resolution`: texto livre em linguagem natural, candidatos a análise de NLP (não a estatística descritiva tradicional).

**Variáveis de data/hora**

- `Date of Purchase`, `First Response Time`, `Time to Resolution` estão como `object` (string), não como `datetime` — precisam de `pd.to_datetime()` antes de qualquer cálculo de prazo/duração.
- `First Response Time` e `Time to Resolution` só existem, respectivamente, para chamados que já receberam resposta e para chamados já resolvidos.


## Verificação da Qualidade dos Dados

In [19]:
import unicodedata

colunas_categoricas = ["Customer Gender", "Product Purchased", "Ticket Type", "Ticket Subject", "Ticket Status", "Ticket Priority", "Ticket Channel"]

print_formatado("1. Valores ausentes:")
print("Valores ausentes (NaN):")
print(df.isnull().sum())

print("\nValores ausentes representados por '?':")
for coluna in df.select_dtypes(include="object").columns:
    qtd_interrogacao = (df[coluna].str.strip() == "?").sum()
    print(f"{coluna:<30}: {qtd_interrogacao}")
print("Ver Markdown")

print_formatado("2. Registros duplicados:")
print("Duplicatas completas:")
print(df.duplicated().sum())

print("\nDuplicadas desconsiderando Ticket ID:")
print(df.drop(columns=["Ticket ID"]).duplicated().sum())
print("Ver Markdown")

print_formatado("3. Tipos de Dados:")
for coluna in df:
    print(df[coluna].apply(type).value_counts())

print("\nQuantidade de categorias por coluna:")
for coluna in df:
    print(f"{coluna:<30}: {df[coluna].nunique()}")

print("\nDistribuição absoluta:")
for coluna in df:
    if df[coluna].dtype != "object":
        continue
    print(df[coluna].value_counts(), "\n")
print("Ver Markdown")


print_formatado("4. Categorias escritas de maneira diferente:")
def normalizar_texto(valor):
    if pd.isna(valor):
        return valor

    valor = valor.strip().lower()
    valor = (
        unicodedata.normalize('NFKD', valor)
        .encode('ASCII', 'ignore')
        .decode('utf-8')
    )

    return valor

for coluna in colunas_categoricas:
    com_normalizacao = df[coluna].apply(normalizar_texto).nunique()
    sem_normalizacao = df[coluna].nunique()
    print(f"{coluna:<30} -> Com Normalização: {com_normalizacao} | Sem Normalização: {sem_normalizacao}")

print_formatado("5. Textos vazios ou espaços extras:")
for coluna in colunas_categoricas:
    qtd_espacos = (df[coluna] != df[coluna].str.strip()).sum()
    print(f"{coluna:<30}: {qtd_espacos}")

print_formatado("6. Valores impossíveis ou fora do contexto:")
print("\nValores impossíveis:")
print(f"Customer Age mínimo                  : {df['Customer Age'].min()}")
print(f"Customer Age máximo                  : {df['Customer Age'].max()}")
print(f"Customer Age < 18 ou > 100            : {df['Customer Age'][(df['Customer Age'] < 18) | (df['Customer Age'] > 100)].count()}")
print(f"Customer Satisfaction Rating mínimo   : {df['Customer Satisfaction Rating'].min()}")
print(f"Customer Satisfaction Rating máximo   : {df['Customer Satisfaction Rating'].max()}")
print(f"Rating fora do intervalo 1-5          : {df['Customer Satisfaction Rating'][(df['Customer Satisfaction Rating'] < 1) | (df['Customer Satisfaction Rating'] > 5)].count()}")

print("\nDatas fora de um intervalo plausível:")
data_compra = pd.to_datetime(df["Date of Purchase"], errors="coerce")
resposta = pd.to_datetime(df["First Response Time"], errors="coerce")
resolucao = pd.to_datetime(df["Time to Resolution"], errors="coerce")
print(f"Date of Purchase mínima               : {data_compra.min()}")
print(f"Date of Purchase máxima               : {data_compra.max()}")
print(f"Datas de compra não convertidas       : {data_compra.isna().sum() - df['Date of Purchase'].isna().sum()}")

print("\nConsistência temporal entre as colunas de data/hora:")
print(f"First Response Time antes de Date of Purchase  : {(resposta < data_compra).sum()}")
print(f"Time to Resolution antes de First Response Time: {(resolucao < resposta).sum()}")

print("\nInconsistência de capitalização (mesmo valor, cases diferentes):")
for coluna in colunas_categoricas:
    valores_lower = df[coluna].str.lower()
    qtd_original = df[coluna].nunique()
    qtd_lower = valores_lower.nunique()
    if qtd_original != qtd_lower:
        print(f"{coluna:<30}: {qtd_original} categorias originais -> {qtd_lower} após lowercase")
    else:
        print(f"{coluna:<30}: sem inconsistência de capitalização")

print("\nCaracteres inesperados em colunas categóricas (fora de letras, espaço, hífen e '/'):")
for coluna in colunas_categoricas:
    padrao = r"[^A-Za-zÀ-ú\s\-/]"
    qtd_estranhos = df[coluna][df[coluna].str.contains(padrao, na=False, regex=True)].nunique()
    valores_estranhos = df[coluna][df[coluna].str.contains(padrao, na=False, regex=True)].unique()
    if qtd_estranhos > 0:
        print(f"{coluna:<30}: {qtd_estranhos} valor(es) -> {list(valores_estranhos)}")
    else:
        print(f"{coluna:<30}: nenhum caractere inesperado")

print("\nConsistência entre 'Ticket Status' e presença dos campos de resolução:")
fechados_sem_resolucao = df[(df["Ticket Status"] == "Closed") & (df["Resolution"].isna())].shape[0]
abertos_com_resolucao = df[(df["Ticket Status"] != "Closed") & (df["Resolution"].notna())].shape[0]
print(f"Tickets 'Closed' sem Resolution preenchida       : {fechados_sem_resolucao}")
print(f"Tickets não 'Closed' com Resolution preenchida   : {abertos_com_resolucao}")



1. Valores ausentes:
Valores ausentes (NaN):
Ticket ID                          0
Customer Name                      0
Customer Email                     0
Customer Age                       0
Customer Gender                    0
Product Purchased                  0
Date of Purchase                   0
Ticket Type                        0
Ticket Subject                     0
Ticket Description                 0
Ticket Status                      0
Resolution                      5700
Ticket Priority                    0
Ticket Channel                     0
First Response Time             2819
Time to Resolution              5700
Customer Satisfaction Rating    5700
dtype: int64

Valores ausentes representados por '?':
Customer Name                 : 0
Customer Email                : 0
Customer Gender               : 0
Product Purchased             : 0
Date of Purchase              : 0
Ticket Type                   : 0
Ticket Subject                : 0
Ticket Description            : 

/var/folders/fq/9y4m8nwn2k7csvmj8sgbvgqr0000gn/T/ipykernel_13371/2679508901.py:10: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  for coluna in df.select_dtypes(include="object").columns:


### Análise da Verificação de Qualidade

**1. Valores ausentes**

Não há valores representados por `'?'` — os nulos aparecem só como `NaN` mesmo, em 4 colunas:

- `Resolution`, `Time to Resolution` e `Customer Satisfaction Rating`: 5.700 nulos cada — exatamente os chamados que **ainda não estão `Closed`**. Não é dado faltante por erro de coleta, é ausência estrutural (esses campos só existem quando o ticket é resolvido).
- `First Response Time`: 2.819 nulos — chamados que ainda não receberam a primeira resposta da equipe (subconjunto dos que não estão `Closed`).

Ou seja, o percentual de nulos deve ser tratado por segmentação (`Ticket Status`), e não por imputação genérica.

**2. Registros duplicados**

Zero duplicatas completas e zero duplicatas mesmo ignorando `Ticket ID` — não há indício de linhas repetidas a remover.

**3. Tipos de dados e distribuição de categorias**

Os tipos por coluna são consistentes (sem mistura de `int`/`str` na mesma coluna). A quantidade de categorias e a distribuição absoluta mostram um dataset **bem balanceado**, típico de dado sintético/gerado:

- `Ticket Status`: `Pending Customer Response` (2.881), `Open` (2.819), `Closed` (2.769) — praticamente um terço cada.
- `Ticket Priority`: entre 2.063 (`Low`) e 2.192 (`Medium`) — as 4 categorias bem próximas.
- `Ticket Channel`: entre 2.073 (`Chat`) e 2.143 (`Email`) — os 4 canais bem próximos.
- `Customer Gender`: `Male` (2.896), `Female` (2.887), `Other` (2.686) — sem desbalanceamento relevante.

Não há classe rara/dominante em nenhuma coluna categórica, o que reduz a preocupação com desbalanceamento para análises futuras, mas também é um sinal de que a distribuição não reflete necessariamente um cenário real de suporte (que tende a ser mais desbalanceado).


## Limpeza dos dados

In [20]:
print_formatado("1. Remover/tratar registros duplicados desnecessários;")
print("\nRemover duplicadas completas:")

print(f"Duplicatas completas: {df.duplicated().sum()}")
print(f"Duplicadas desconsiderando Ticket ID: {df.drop(columns=['Ticket ID']).duplicated().sum()}")

print("Não há duplicatas para remover")


print_formatado("2. Remover/tratar valores ausentes de forma justificada;")

print("Valores ausentes antes do tratamento:")
nulos = df.isnull().sum()
print(nulos[nulos > 0])

print("\nVerificação da origem dos nulos:")
print(f"{'Tickets não Closed':<40}: {(df['Ticket Status'] != 'Closed').sum()}")
print(f"{'Nulos em Resolution':<40}: {df['Resolution'].isna().sum()}")
print(f"{'Nulos em Time to Resolution':<40}: {df['Time to Resolution'].isna().sum()}")
print(f"{'Nulos em Customer Satisfaction Rating':<40}: {df['Customer Satisfaction Rating'].isna().sum()}")
print(f"{'Tickets Open':<40}: {(df['Ticket Status'] == 'Open').sum()}")
print(f"{'Nulos em First Response Time':<40}: {df['First Response Time'].isna().sum()}")
print("Ver Markdown")


print_formatado("3. Corrigir tipos das colunas:")

colunas_data = ["Date of Purchase", "First Response Time", "Time to Resolution"]
prioridades_ordenadas = ["Low", "Medium", "High", "Critical"]

print("Tipos antes da conversão:")
print(df[colunas_data + ["Customer Satisfaction Rating"] + colunas_categoricas].dtypes)

for coluna in colunas_data:
    df[coluna] = pd.to_datetime(df[coluna], errors="coerce")

df["Customer Satisfaction Rating"] = df["Customer Satisfaction Rating"].astype("Int8")

for coluna in colunas_categoricas:
    if coluna == "Ticket Priority":
        df[coluna] = pd.Categorical(df[coluna], categories=prioridades_ordenadas, ordered=True)
    else:
        df[coluna] = df[coluna].astype("category")

print("\nTipos depois da conversão:")
print(df[colunas_data + ["Customer Satisfaction Rating"] + colunas_categoricas].dtypes)


print_formatado("4. Corrigir espaços, erros de preenchimento e inconsistências evidentes;")

print("Não foi verificado na etapa de qualidade espaços extras para corrigir.")

qtd_placeholder = df["Ticket Description"].str.contains("{product_purchased}", regex=False).sum()
print(f"\nDescrições com o placeholder '{{product_purchased}}' antes : {qtd_placeholder}")

df["Ticket Description"] = [
    descricao.replace("{product_purchased}", produto)
    for descricao, produto in zip(df["Ticket Description"], df["Product Purchased"])
]

qtd_placeholder = df["Ticket Description"].str.contains("{product_purchased}", regex=False).sum()
print(f"Descrições com o placeholder '{{product_purchased}}' depois: {qtd_placeholder}")
print("Ver Markdown")


print_formatado("5. Corrigir valores inválidos;")

invalidos = df["Time to Resolution"] < df["First Response Time"]
print(f"Tickets com Time to Resolution anterior ao First Response Time antes : {invalidos.sum()}")

df.loc[invalidos, "Time to Resolution"] = pd.NaT

print(f"Tickets com Time to Resolution anterior ao First Response Time depois: {(df['Time to Resolution'] < df['First Response Time']).sum()}")
print(f"Tickets 'Closed' com Time to Resolution nulo após o tratamento       : {((df['Ticket Status'] == 'Closed') & (df['Time to Resolution'].isna())).sum()}")
print("Ver Markdown")



1. Remover/tratar registros duplicados desnecessários;

Remover duplicadas completas:
Duplicatas completas: 0
Duplicadas desconsiderando Ticket ID: 0
Não há duplicatas para remover


2. Remover/tratar valores ausentes de forma justificada;
Valores ausentes antes do tratamento:
Resolution                      5700
First Response Time             2819
Time to Resolution              5700
Customer Satisfaction Rating    5700
dtype: int64

Verificação da origem dos nulos:
Tickets não Closed                      : 5700
Nulos em Resolution                     : 5700
Nulos em Time to Resolution             : 5700
Nulos em Customer Satisfaction Rating   : 5700
Tickets Open                            : 2819
Nulos em First Response Time            : 2819
Ver Markdown


3. Corrigir tipos das colunas:
Tipos antes da conversão:
Date of Purchase                    str
First Response Time                 str
Time to Resolution                  str
Customer Satisfaction Rating    float64
Customer Ge

## Observações Limpeza dos Dados

### 1. Remover/tratar registros duplicados desnecessários
* Não há duplicatas completas nem duplicatas desconsiderando `Ticket ID`.
    * Nenhuma remoção necessária.

### 2. Remover/tratar valores ausentes de forma justificada
* Os nulos são estruturais: `Resolution`, `Time to Resolution` e `Customer Satisfaction Rating` (5.700) correspondem exatamente aos tickets não `Closed`; `First Response Time` (2.819) aos tickets `Open`.
    * Manter os `NaN`: imputar criaria dados que não existem e remover as linhas descartaria ~67% do dataset.
        * Análises desses campos devem filtrar por `Ticket Status` (ex.: satisfação média apenas entre `Closed`).

### 3. Corrigir tipos das colunas
* Converter `Date of Purchase`, `First Response Time` e `Time to Resolution` de `object` para `datetime64`.
* Converter `Customer Satisfaction Rating` de `float64` para `Int8`: a nota é discreta de 1 a 5, só era float por causa dos nulos.
* Converter as colunas categóricas para o tipo `category`, com `Ticket Priority` ordenada (`Low < Medium < High < Critical`).

### 4. Corrigir espaços, erros de preenchimento e inconsistências evidentes
* Todas as 8.469 descrições de `Ticket Description` contêm o placeholder `{product_purchased}` sem substituição.
    * Substituir o placeholder pelo valor de `Product Purchased` da própria linha.

### 5. Corrigir valores inválidos
* 1.365 tickets com `Time to Resolution` anterior ao `First Response Time` (resolução antes da primeira resposta).
    * Converter o `Time to Resolution` desses registros para `NaT`, pois não há como recuperar o valor real.
        * Esses tickets são `Closed` e ficam sem data de resolução; cálculos de duração passam a usar apenas o subconjunto com timestamp válido.

### Outras observações:
* Não há espaços extras, diferenças de capitalização ou acentuação a corrigir nas colunas categóricas.
* Não há valores fora de faixa em `Customer Age` (18 a 70) nem em `Customer Satisfaction Rating` (1 a 5).
* As datas de compra estão em intervalo plausível (2020-01-01 a 2021-12-30) e nenhuma falhou na conversão.
* O placeholder não substituído e a inconsistência temporal reforçam que o dataset é sintético.
